plugins/modules/proxmox_template.py

```diff
@@ -303,6 +357,9 @@ def main():
                 elif not proxmox.delete_template(node, storage, content_type, template, timeout):
                     module.fail_json(changed=False, msg='failed to delete template with volid=%s:%s/%s' % (storage, content_type, template))
 
+            if verify_checksum:
+                if proxmox.verify_checksum(node, storage, url, content_type, timeout, checksum, checksum_algorithm):
+                    module.exit_json(changed=True, msg="Checksum verified, template with volid=%s:%s/%s uploaded" % (storage, content_type, template))
```

Comment:
if `verify_checksum` is `True`, it does not do anything else? That would force the user to make a second call to get things done. Perhaps it would be better to go with:
```suggestion
            if verify_checksum:
                if not proxmox.verify_checksum(node, storage, url, content_type, timeout, checksum, checksum_algorithm):
                    module.fail_json(changed=True, msg="Checksum failed to verify, template with volid=%s:%s/%s uploaded" % (storage, content_type, template))
```